src/main/java/swerve/ModuleIOReal.java

```diff
@@ -110,47 +111,59 @@ public void updateInputs(SwerveModuleInputs inputs) {
 
     @Override
     public double getAngle() {
-        return AngleUtil.normalize(ticksPerRad.toUnits(angleMotor.getSelectedSensorPosition()));
+        return AngleUtil.normalize(Units.rotationsToRad(angleMotor.getPosition().getValue()));
     }
 
     @Override
     public void setAngle(double angle) {
         angleSetpoint = AngleUtil.normalize(angle);
         Rotation2d error = new Rotation2d(angle).minus(new Rotation2d(currentAngle));
-        angleMotor.set(TalonFXControlMode.MotionMagic, angleMotorPosition + ticksPerRad.toTicks(error.getRadians()));
+        var motorRequest = new MotionMagicVoltage(angleMotor.getPosition().getValue() + error.getRotations());
+        angleMotor.setControl(motorRequest);
     }
 
     @Override
     public double getVelocity() {
-        return ticksPerMeter.toVelocity(driveMotor.getSelectedSensorVelocity());
+        return driveMotor.getVelocity().getValue();
     }
 
     @Override
     public void setVelocity(double velocity) {
         var angleError = new Rotation2d(angleSetpoint).minus(new Rotation2d(currentAngle));
         velocity *= angleError.getCos();
+
         driveMotorVelocitySetpoint = velocity;
-        driveMotor.set(TalonFXControlMode.Velocity, ticksPerMeter.toTicks100ms(velocity));
+        var motorRequest = new VelocityDutyCycle(
+                Units.metersPerSecondToRps(
+                        velocity,
+                        SwerveConstants.WHEEL_DIAMETER / 2
+                )
+        );
+        driveMotor.setControl(motorRequest);
     }
 
     @Override
     public SwerveModulePosition getModulePosition() {
         return new SwerveModulePosition(
-                ticksPerMeter.toUnits(driveMotor.getSelectedSensorPosition()),
+                Units.rpsToMetersPerSecond(
+                        driveMotor.getPosition().getValue(),
```

Comment:
Something's weird here, the function you use receives a velocity unit, but you pass it a distance unit.

Reply:
It takes rotations per second and turns it into meters per second, it's the same calculation for turning rotations into meters.

Reply:
It's still confusing. Instead, create two different functions, or just rename the existing one to be more general (no need to mention the "per second").